lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -105,6 +105,18 @@ static void foreachKernelArgMD(
   }
 }
 
+static SPIRVMemoryModelKind getMemoryModel(Module &M) {
+  NamedMDNode *MemoryModelMD = M.getNamedMetadata(kSPIRVMD::MemoryModel);
+  if (MemoryModelMD) {
+    auto Model = static_cast<SPIRVMemoryModelKind>(
```

Comment:
I think we should either return result of static cast directly, or break down this expression into several simpler ones, possibly with checks that pointers are not null (static analysis tools may find issues here).

Reply:
I added some checks